astropy/coordinates/representation.py

```diff
@@ -83,8 +84,13 @@ class BaseRepresentation(ShapedLikeNDArray):
     that frame classes may override this with their own preferred units.
     """
 
+    attr_classes = OrderedDict()
+
     recommended_units = {}  # subclasses can override
 
+    # Ensure multiplication/division with ndarray doesn't lead to object arrays.
+    __array_priority__ = 50000
```

Comment:
Really all this needs to do is to ensure any `ndarray` subclass has lower priority. Of course, I don't know what the highest possible one is...